### PR TITLE
fix(Rebuild): A file should not be attempted to be deleted when turned into a directory

### PR DESCRIPTION
**Pull request by MikePresman**
Resolves GGT-3870

### CODE REVIEW

File: test/client_rebuild_test.go [hunk 1]
```diff
@@ -438,3 +438,37 @@ func TestRebuildWithMissingMetadataDir(t *testing.T) {
...
 	})
 }
+
+func TestRebuildFileBecomesADir(t *testing.T) {
+	tc := util.NewTestCtx(t, auth.Project, 1)
+	defer tc.Close()
+
+	writeProject(tc, 1, 2)
+	writeObject(tc, 1, 1, i(2), "a.html", "a v1")
+	writeObject(tc, 1, 2, nil, "a.html/foo", "a v2")
```
Comment (scott-rc) [inline test/client_rebuild_test.go:448]: Could we do this instead?

```go
	writeObject(tc, 1, 1, i(2), "a.html", "a v1")
	writeObject(tc, 1, 2, nil, "a/b.html", "a v2")
```

This ☝️ feels like a more realistic example.

Comment (angelini) [reply test/client_rebuild_test.go:448]: That will not trigger the error. You need to turn a file into a directory of the same name

**Review by angelini — APPROVED** [review]
This LGTM, but let's generate a prod query and run an `EXPLAIN ANALYZE` against a real project in prod and throw the results here.

File: test/client_rebuild_test.go (hunk 1 above)
Comment (scott-rc) [reply test/client_rebuild_test.go:448]: ah I see! kk 👍

**Pull request merged**